src/browser/setupWorker/start/createRequestListener.ts

```diff
@@ -30,6 +30,8 @@ export const createRequestListener = (
     const request = parseWorkerRequest(message.payload)
     const requestCloneForLogs = request.clone()
 
+    context.requests.set(requestId, request.clone())
```

Comment:
This also means +1 request clone per request.

Reply:
A few important notes:

1. This is always called _once_ per 1 request. 
2. Request resolution hasn't started yet so there's no cache. 

We can technically make this the earliest clone to prevent the first request handler from cloning this request _again_. I think this makes sense. Wdyt, @mattcosta7?

Reply:
I've added this first request clone to `RequestHandler.cache` so no request handlers would have to clone the request. 

It's still a bit of an overhead since now we are _always_ making 1 clone for every outgoing request (in the browser). Previously, only the first request handler would make the clone. I'd say this is not a concern since request cloning was done in the parsing phase, and the only scenario when this actually changes anything is if you have 0 request handlers (than, indeed, no cloning is needed). 

But we still have to clone the request even in that scenario so you can access it in the `response:*` life-cycle events.

Reply:
Yes, I think this is definitely worth it, and the cost seems low enough that it shouldn't be a concern

Reply:
Now request handlers won't clone requests at all, I think. Any outgoing request will be cloned here, and then request handlers will reuse the clone from cache.